shelley/chain-and-ledger/executable-spec/test/STSTests.hs

```diff
@@ -61,17 +61,25 @@ testCHAINExample (CHAINExample slotNow initSt block expectedSt) =
     applySTS @CHAIN (TRC (slotNow, initSt, block)) @?= Right expectedSt
 
 testCHAINExample1 :: Assertion
-testCHAINExample1 = testCHAINExample ex2
+testCHAINExample1 = testCHAINExample ex1
```

Comment:
I wonder whether you could use the `Control.State.Transition.Trace` `.-` and `.->` combinators here. Then you could rewrite these as follows:

```haskell
checkTrace env @CHAIN $ pure st .- sig .-> st'
```

Reply:
And you could even chain applications:

```haskell
checkTrace env @CHAIN $ pure st .- sig0 .-> st1
                                .- sig1 .-> st2
                                .- sig2 .-> st3
```

Reply:
Just a suggestion while I wait for my other tests to finish :grin:

Reply:
yes! that's fantastic!! thanks @dnadales !

Reply:
I just tried it, it works great! I will add this to my next PR / example

Reply:
for chaining them, I will need to think more about how to handle the environment changing, ie `slotNow`.

Reply:
Right, if you need to change the environment you probably don't want to chain them (since a trace assumes a constant environment).